### ISSUE TITLE
Extending config with additional plugins

### ISSUE THREAD

**Issue by niksy** [stylelint/stylelint#1585]
I don’t know if this is correct behavior, but when you try to extend existing config, plugins array of config you are extending is not applied to extended config.

#### .stylelintrc

```json
{
	"extends": [
		"stylelint-config-foo/extended"
	]
}
```

#### test.css

```css
body {
	border:none;
	padding:auto;
}
```

#### index.js

```js
module.exports = {
	plugins: [
		'stylelint-value-border-zero'
	],
	rules: {
		'value-border-zero': [{ convention: '0' }]
	}
};
```

#### extended.js

```js
module.exports = {

	extends: [
		'./'
	].map(require.resolve),

	plugins: [
		'stylelint-value-no-padding-auto'
	],

	rules: {
		'value-no-padding-auto': true
	}
};
```

If you run `stylelint test.css` you will get following error:

```sh
Error: Undefined rule "value-border-zero"
    at exports.default (/Users/niksy/Projects/stylelint-test/node_modules/stylelint/dist/utils/configurationError.js:8:13)
    at /Users/niksy/Projects/stylelint-test/node_modules/stylelint/dist/postcssPlugin.js:113:47
    at Array.forEach (native)
    at /Users/niksy/Projects/stylelint-test/node_modules/stylelint/dist/postcssPlugin.js:111:33
npm ERR! Test failed.  See above for more details.
```

## Current fix

My current fix is to import original config and concatenate plugins array:

```js
module.exports = {

	extends: [
		'./'
	].map(require.resolve),

	plugins: [].concat(require('./').plugins, [
		'stylelint-value-no-padding-auto'
	]),

	rules: {
		'value-no-padding-auto': true
	}
};
```

But this seems kind of counter intuitive.

Is this proper behavior? I suppose internals are using standard object merging which doesn’t mutate arrays.

**Comment by jeddy3** [comment]
@niksy Thanks for the detailed write-up.

This [doesn't seem to be](https://github.com/stylelint/stylelint/pull/1590) something we're accounted for. @stylelint/core Shall we make this change in `7.0`?

 > I suppose internals are using standard object merging which doesn’t mutate arrays.

That appears to be so. Perhaps using [`mergeWith`](https://lodash.com/docs#mergeWith) with a `customizer`, rather than `merge` will help here?

P.S. That's a really nice [stylelint config](https://github.com/niksy/stylelint-config-niksy) you have! :)

**Comment by ntwb** [comment]
> This doesn't seem to be something we're accounted for. @stylelint/core Shall we make this change in 7.0?

SGTM

> P.S. That's a really nice stylelint config you have! :)

Nice 👍

**Comment by jeddy3** [comment]
Done in https://github.com/stylelint/stylelint/pull/1590
